v-next/hardhat/src/internal/builtin-plugins/solidity-test/task-action.ts

```diff
@@ -1,43 +1,68 @@
 import type { RunOptions } from "./runner.js";
 import type { TestEvent } from "./types.js";
+import type { BuildOptions } from "../../../types/solidity/build-system.js";
 import type { NewTaskActionFunction } from "../../../types/tasks.js";
 
 import { finished } from "node:stream/promises";
 
+import { getAllFilesMatching } from "@ignored/hardhat-vnext-utils/fs";
 import { createNonClosingWriter } from "@ignored/hardhat-vnext-utils/stream";
 
-import { getArtifacts, isTestArtifact } from "./helpers.js";
+import { shouldMergeCompilationJobs } from "../solidity/build-profiles.js";
+import {
+  getArtifacts,
+  throwIfSolidityBuildFailed,
+} from "../solidity/build-results.js";
+
+import { getTestSuiteIds } from "./helpers.js";
 import { testReporter } from "./reporter.js";
 import { run } from "./runner.js";
 
 interface TestActionArguments {
   timeout: number;
-  noCompile: boolean;
+  force: boolean;
+  quiet: boolean;
 }
 
 const runSolidityTests: NewTaskActionFunction<TestActionArguments> = async (
-  { timeout, noCompile },
+  { timeout, force, quiet },
```

Comment:
I decided to remove the `noCompile` flag from here until we figure out the right semantics for it. 

I also added `force` and `quiet` flags as they are present in the compile task and make sense here as well.